### ISSUE TITLE
Higher order gradients with Instance Normalization crashes unless `.backward(retain_graph=True)`

### ISSUE THREAD

**Issue by srxdev0619** [facebookresearch/higher#3]
Hi,

Thanks so much for creating this library! 

I've run into problems when using `nn.InstanceNorm2d` within networks. More specifically, in the [MAML-Omniglot example](https://github.com/facebookresearch/higher/blob/master/examples/maml-omniglot.py) replacing the `nn.BatchNorm2d` with `nn.InstanceNorm2d` as follows

```python
net = nn.Sequential(
        nn.Conv2d(1, 64, 3),
        nn.InstanceNorm2d(64, affine=True),
        nn.ReLU(inplace=True),
        nn.MaxPool2d(2, 2),
        nn.Conv2d(64, 64, 3),
        nn.InstanceNorm2d(64, affine=True),
        nn.ReLU(inplace=True),
        nn.MaxPool2d(2, 2),
        nn.Conv2d(64, 64, 3),
        nn.InstanceNorm2d(64, affine=True),
        nn.ReLU(inplace=True),
        nn.MaxPool2d(2, 2),
        Flatten(),
        nn.Linear(64, args.n_way)).to(device) 
```

Gives me the following error:

`RuntimeError: Trying to backward through the graph a second time, but the buffers have already been freed. Specify retain_graph=True when calling backward the first time.`

This error is fixed if line 162 in `maml-omniglot.py` is changed to `qry_loss.backward(retain_graph=True)`, but the program now consumes significantly more GPU memory.

It is not immediately clear to me why `nn.InstanceNorm2d` should require `retain_graph=True` at last call of `.backward()`. Perhaps this is worth looking into?

Note: I'm not sure if this helps, but I've encountered the same problem when using  [Adam Paszke
's](https://gist.github.com/apaszke) (great!) [gist](https://gist.github.com/apaszke/4c8ead6f17a781d589f6655692e7f6f0) to create functional versions of modules.

Thank you!

**Comment by bamos** [comment]
Hmm, I just tried running this on my system with the batch norms replaced by instance norms and it's working. What PyTorch version are you using? I'm on:

```
$ python3 -c 'import torch; print(torch.__version__)'
1.3.0.dev20190907
```

**Comment by bamos** [comment]
Also for the sake of debugging, if you are on the latest version of PyTorch, can you also try replacing the `net.train` calls with `net.eval` calls just to see if this works?

**Comment by srxdev0619** [comment]
I'm using 
```
$ python -c 'import torch; print(torch.__version__)'
1.2.0
```
And yes, I get the same error if I replace all `net.train()` calls with `net.eval()` using the above version. I'll rerun the same with  pytorch 1.3.0 to see if I can reproduce it there.

**Comment by srxdev0619** [comment]
This issue seems to be fixed in `1.3.0.dev20190917`.  Thanks!

**Comment by egrefen** [comment]
Okay sounds good. We'll update the requirements in the README.
